src/lib/parse/headers.js

```diff
@@ -0,0 +1,117 @@
+"use strict"
+//@flow
+
+const types = require('./types-checker');
+
+export type HeaderDef = {
+    name: string,
+    value: string,
+    type: string,
+    required?: boolean,
+};
+type Match = {
+    groups: MatchGroups
+};
+
+type MatchGroups = {
+    required: string,
+    value: string,
+    type: string
+};
+
+export type HeaderValidation = {
+    valid: boolean,
+    messages: Array<string>
+};
+
+const regExp = /`?(?<value>[^`]*?)`?\s*\((?<type>[^,]*),?\s?(?<required>.*)\)/;
+
+const parseHeaderValue = (rawHeader: HeaderDef): HeaderDef => {
+    // $FlowFixMe - flow does not support match group names
+    let match: ?Match = regExp.exec(rawHeader.value);
+    if (match) {
+        let required: boolean;
+        switch (match.groups.required) {
+            case 'required':
+            case '':
+                required = true;
+                break;
+            case 'optional':
+                required = false;
+                break;
+            default:
+                throw new Error(`For header "${rawHeader.name}", unrecognized optionality: "${match.groups.required}"`);
+        }
+        return {
+            name: rawHeader.name,
+            value: match.groups.value,
+            type: match.groups.type,
+            required: required,
+        };
+    }
+
+    return {
+        name: rawHeader.name,
+        value: rawHeader.value,
+        type: '',
+        required: true,
+    };
+};
+
+const compareFixtureAndContractHeaders = (fixtureHeaders: ?Array<HeaderDef>, contractHeaders: Array<HeaderDef>): HeaderValidation => {
+    fixtureHeaders = fixtureHeaders || [];
+    const result = {
+        valid: true,
+        messages: [],
+    };
+
+    const fixtureMap = fixtureHeaders.reduce((previous, current) => {
+        previous[current.name] = current;
```

Comment:
consider renaming `previous` to `accumulator` or `map` and `current` to `item` to make it clear we are converting an array into a map